### DISCUSSION TITLE
What is the proper way to serialize Nodes?

### DISCUSSION THREAD

**Discussion by ruleBased848** [bevyengine/bevy#12700] [Q&A] [answered]
I am looking for the proper way to save my UIs as a RON file.
I followed [this](https://github.com/bevyengine/bevy/blob/latest/examples/scene/scene.rs) example, and it worked perfectly.
Then, I add a line:
```rust
...
let mut component_b = ComponentB::from_world(world);
component_b.value = "hello".to_string();
scene_world.spawn((
    component_b,
    ComponentA { x: 1.0, y: 2.0 },
    Transform::IDENTITY,
    Name::new("joe"),
));
scene_world.spawn(ComponentA { x: 3.0, y: 4.0 });
scene_world.insert_resource(ResourceA { score: 1 });
scene_world.spawn(NodeBundle::default()); // I added this.
...
```
It did compile, but when I ran it, it panicked with message "called `Result::unwrap()` on an `Err` value: Message("Type 'core::num::NonZeroU16' did not register ReflectSerialize")".
Does it mean that nodes are not able to be serialized into RON format? So I can't serialize my UIs?
I believe I am just missing something. Many thanks!

**Comment by MrGVSV** [comment, marked as answer]
Sounds like something in `NodeBundle` uses `NonZeroU16`, and it's saying that it's not currently registered. Try registering it like:

```rust
app.register_type::<NonZeroU16>();
```

Note that in 0.14 this won't need to be done manually anymore as we will have recursive registration (#5781).
